### ISSUE TITLE
In index resource list, put export facets in parameter and change client requests

### ISSUE THREAD

**Issue by luis100** [keeps/roda#562]
Exporting results or facets should be a REST API option

**Comment by rui-castro** [comment]
@luis100 can you be more specific, please?

Can you use [these examples](https://github.com/keeps/roda/issues/319#issuecomment-248980055) to specify what you want.

**Comment by luis100** [comment]
The option to export in CSV the index results or the index facet results was made selecting the facet results if facets were used. This is not what was required when we are exporting lists that actually have facets.

So we've already implemented this for the backend and POST methods. There is a new argument (boolean exportFacets) in the backend methods and POST methods that allows to select if we want to export the results or the facet results.

But this is not done in the GET method. We maintained the previous behaviour (i.e. it will export facets results if facets are not empty), but maybe this should be reviewed.
